test/e2e/specs/editor/various/a11y.spec.js

```diff
@@ -38,13 +38,24 @@ test.describe( 'a11y (@firefox, @webkit)', () => {
 		// Navigate to the 'Editor top bar' region.
 		await pageUtils.pressKeys( 'ctrl+`' );
 
-		// This test assumes the Editor is not in Fullscreen mode. Check the
-		// first tabbable element within the 'Editor top bar' region is the
-		// 'Block Inserter' button.
 		await pageUtils.pressKeys( 'Tab' );
-		await expect(
-			page.locator( 'role=button[name=/Block Inserter/i]' )
-		).toBeFocused();
+
+		const isFullScreenMode = await page.evaluate( () => {
+			return window.wp.data
+				.select( 'core/edit-post' )
+				.isFeatureActive( 'fullscreenMode' );
+		} );
+		// When full screen mode is enabled, check the first tabbable element
+		// within the 'Editor top bar' region is the 'View Posts' link otherwise
+		// check it's the 'Block Inserter' button.
+		// When running this test locally, make sure that in macOS webkit (Safari)
+		// links are focusable and tabbable by setting the related preferences at
+		// system and browser level.
+		const elementToTest = isFullScreenMode
+			? 'role=link[name=/View Posts/i]'
+			: 'role=button[name=/Block Inserter/i]';
+
```

Comment:
If it's guaranteed that the e2e tests run with fullscreen mode always enabled, this condition may be unnecessary .